fe/fe-core/src/main/java/com/starrocks/planner/SplitCastDataSink.java

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.planner;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.starrocks.analysis.Expr;
+import com.starrocks.thrift.TDataSink;
+import com.starrocks.thrift.TDataSinkType;
+import com.starrocks.thrift.TDataStreamSink;
+import com.starrocks.thrift.TExplainLevel;
+import com.starrocks.thrift.TExpr;
+import com.starrocks.thrift.TPlanFragmentDestination;
+import com.starrocks.thrift.TSplitDataStreamSink;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SplitCastDataSink extends DataSink {
+    // SplitCastDataSink will have multiple dest fragment
+    private final List<DataStreamSink> dataStreamSinks = Lists.newArrayList();
```

Comment:
At present, only two DataStreamSink, one for random while the other for shuffle, in future, is there a possiblity that more than two dataStreamSinks are introduced to support other scenarios?

Reply:
> At present, only two DataStreamSink, one for random while the other for shuffle, in future, is there a possiblity that more than two dataStreamSinks are introduced to support other scenarios?

right now, left table 's exchange Node has two DataStreamSink, one for  random while the other for shuffle. and right table's exchange Node  also has two DataStreamSink, one for  broadcast  while the other for shuffle. you can define what outputPartition you want, it's very flexible